set-mongo/src/main/java/eu/europeana/set/mongo/service/PersistentUserSetServiceImpl.java

```diff
@@ -89,6 +92,26 @@ public PersistentUserSet getByIdentifier(String identifier) {
 		return getUserSetDao().findOne(query);
 	}
 	
+	/* (non-Javadoc)
+	 * @see eu.europeana.set.mongo.service.PersistentUserSetService#isTypeAndCreatorExisting(java.lang.String, eu.europeana.set.definitions.model.agent.Agent)
+	 */
+	public boolean isTypeAndCreatorExisting(String type, Agent creator) {
+	    Query<PersistentUserSet> query = getUserSetDao().createQuery();
+	    query.filter(PersistentUserSet.FIELD_TYPE, type);
```

Comment:
the filter for creatorId must be added to the Query must be added to the

Reply:
It is not clear how to do that because, crator is an object with subfields. Is there any example for that?

Reply:
Added additional filter for crator according to mongo specs

Reply:
Refactored to return user set object using type constant